### DISCUSSION TITLE
Ability to Index from within the app (For SMB)

### DISCUSSION THREAD

**Discussion by Tailsproxy** [Radiokot/photoprism-android-client#46] [Ideas]
Thanks again for all your hard work on this app, it is really exciting to see the app fleshing out and so fast!

I backup my photos from my phone via SMBShare2 and PhotoPrism does not display these photos in a users library until the user manually indexes them.  I understand you do not want to copy features of the web app or directly manage syncing, I was however wondering if you had considered adding the ability to trigger PP to index from within your app.  Currently it is a bit cumbersome to index as I have to log in via a browser, (or by clicking on a photo in your app and opening the web app), navigate to my 'Library' and then hit 'Start'. I tend to end up a few days or weeks behind in my photo's which somewhat reduces the moment to moment usefulness of the app when I just want to quickly show a friend a photo or check a screenshot. 

It would make the app feel more complete if I could hit a button or setting to run this index to more easily keep my library up to date.

**Comment by ippocratis** [comment]
This is not what you ask but it will save you time and it will automate your workflow.
Photoprism has a feature called import
You specify an import folder where you move your assets before they added to your originals.
If you use webdav to add files to that folder,  an import job will be iniciated after a specific  time period automatically.
You can also choose to move files from import to originals not just copy them.
So you could change from smb to webdav 
Android webdav sync clients like foldersync and photosync are also support mtls client cert auth if you secured your photoprism server with that option.

https://docs.photoprism.app/user-guide/library/import/

**Comment by Radiokot** [comment]
Hey!
Have you tried playing around with PhotoPrism auto import configuration? I have it enabled, my photos are uploaded directly to the `originals` subfolder and PhotoPrism launches the indexer by itself. The configuration param is `PHOTOPRISM_AUTO_INDEX`, mine is set to 30 seconds after changes occurred: `PHOTOPRISM_AUTO_INDEX: 30`
